.github/workflows/release-artifacts.yml

```diff
@@ -116,18 +122,29 @@ jobs:
       - name: Install cibuildwheel
         run: python -m pip install cibuildwheel==2.9.0 poetry==1.2.0
 
-      # Only build a single wheel in CI.
-      - name: Set env vars.
-        run: |
-          echo "CIBW_BUILD="cp37-manylinux_x86_64"" >> $GITHUB_ENV
+      - name: Set up QEMU to emulate aarch64
+        if: matrix.arch == 'aarch64'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+
+      - name: Build aarch64 wheels
+        if: matrix.arch == 'aarch64'
+        run: echo 'CIBW_ARCHS_LINUX=aarch64' >> $GITHUB_ENV
```

Comment:
I wonder if it would be clearer to unconditionally `echo 'CIBW_ARCHS_LINUX=${{ matrix.arch }}' >> $GITHUB_ENV` in the "Build wheels" step?

Reply:
That would exclude `i686` wheels on release, which are currently included. So the one or the other way round (adding `i686` to x86 CIBW env), an extra step is necessary. From: https://github.com/matrix-org/synapse/actions/runs/3273187715/jobs/5385086794
```
environment: {'assignments': [PATH="$PATH:$HOME/.cargo/bin"]}
    cp[37](https://github.com/matrix-org/synapse/actions/runs/3273187715/jobs/5385086794#step:6:38)-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp[38](https://github.com/matrix-org/synapse/actions/runs/3273187715/jobs/5385086794#step:6:39)-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp[39](https://github.com/matrix-org/synapse/actions/runs/3273187715/jobs/5385086794#step:6:40)-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp310-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp311-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp37-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp38-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp39-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp310-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp311-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp37-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp38-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp39-manylinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp37-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp38-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    pp39-manylinux_i686: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp37-musllinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp38-musllinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp39-musllinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp310-musllinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
    cp311-musllinux_x86_64: ParsedEnvironment(['PATH="$PATH:$HOME/.cargo/bin"'])
```
Just in theory, the `aarch64` container should be able to build `armv7l` wheels as well and alternatively an `arm` container is available with CIBW. But a task for another PR, if wanted 😉.

Reply:
@erikjohnston erm, did we intend to build wheels for i686? Do we even support 32-bit platforms?

Reply:
They are all here: https://pypi.org/project/matrix-synapse/#files
So I guess you would have received reports if they did not work at all 😉.

Reply:
I have no `i686` system for testing, but 32-bit in general is not an issue as Synapse works on ARM 32-bit. We offer it via DietPi OS for `armv6l` (RPi) and `armv7l` systems for a long time as install option (via `pip`) and piwheels.org shipped wheels as well until recent Rust addition broke builds (only because the build system Rust is too old, AFAIK, fix on the way): https://piwheels.org/project/matrix-synapse/